src/index.ts

```diff
@@ -299,6 +291,10 @@ const plugin: JupyterFrontEndPlugin<void> = {
     /**
      * Add custom Share notebook command
      */
+    const markManualSave = () => {
+      hasManuallySaved = true;
```

Comment:
This assumes only one notebook can be edited in the session. Is this correct?

I do not want to drag the review on this PR but I consider the multiple callabacks being added a code smell for state management.

Could possibly some transient state be added to notebook metadata? Maybe "lastSavedInSession" where session is a unique UUID of the current and then you could compute `hasManuallySaved` by checking if matches the current session ID?